src/components/DefaultContainer/DefaultContainer.tsx

```diff
@@ -126,14 +123,12 @@ export const DefaultContainer = withRouter(props => {
         .then(response => {
           if (response.isRight()) {
             const respValue = response.value;
-            if (respValue.status < 300) {
-              if (UserProfile.is(respValue.value)) {
-                const userProfileResp = respValue.value;
-                handleGetUserProfile(userProfileResp);
-                if (!userProfileResp.work_email) {
-                  toggleAddMailModal();
-                }
-              }
+            if (respValue.status === 200) {
+              const userProfileResp = respValue.value;
```

Comment:
here response.value is the correct type (UserProfile), we don't need to check it